django/db/models/expressions.py

```diff
@@ -1756,6 +1756,7 @@ def __init__(self, queryset, output_field=None, **extra):
         # Allow the usage of both QuerySet and sql.Query objects.
         self.query = getattr(queryset, "query", queryset).clone()
         self.query.subquery = True
+        self.template = extra.pop("template", self.template)
```

Comment:
I deliberately avoided adding tests for `extra`, given that discussion in ticket-35459 is ongoing. There is of course implicit coverage that a `self.template` attribute exists.